### ISSUE TITLE
Clarification on some points in the README for the deployer

### ISSUE THREAD

**Issue by pbkeating** [reconhub/deployer#24]
@zkamvar 

Just installing deployer on my computer.
Unclear from the README whether you must remove any previously existing R or RStudio from your computer if you want to use the deployer.  May be worth highlighting this.

Also, under the section "Installing packages using the deployer", the user needs to open the **open-deployer.rproj** before activating the **activate.r** script, right? Or at least set the working directory to be same as where the **activate.r** script is located?

Also, once the deployer is installed, does the user need to source the activate.r script every time they open a new R session or write a new R script/markdown they do or is it a one time only situation? If yes, may be worth highlighting this in the README also

**Comment by zkamvar** [comment]
Hi @pbkeating,

Thank you for the comments! I agree that we should definitely clarify these areas and will create a PR soon to fix these.

> Unclear from the README whether you must remove any previously existing R or RStudio from your computer if you want to use the deployer. May be worth highlighting this.

If you have administrative rights on your computer, this will replace any existing installations of R and RStudio. You shouldn't need to remove them as far as I can tell.

> Also, under the section "Installing packages using the deployer", the user needs to open the **open-deployer.rproj** before activating the **activate.r** script, right? Or at least set the working directory to be same as where the **activate.r** script is located?

Yes, this is correct. 


> Also, once the deployer is installed, does the user need to source the activate.r script every time they open a new R session or write a new R script/markdown they do or is it a one time only situation? If yes, may be worth highlighting this in the README also

They do not need to source `activate.R` script every time they want to use R. Once they have installed R and all the packages they need from the deployer, then they should be good to go. Is there any more clarification I can give on this?

**Comment by pbkeating** [comment]
Hi Zhian,

Thanks for that.

As for setting the working directory for the activate.R script, I tried to use the open_deployer.rproj, but it didn't open the project and I ended up setting the working directory outside of a project. Normal?

Beyond that, wonder if possible to add the **kableExtra** R package to the list of included packages? Or is there an alternative package recommended for creating tables in the deployer?

In regards to the activate.R script, do I download all the packages I'll need while in that working directory, or should they normally install from within any other script in other working directories after sourcing the activate.R?

Thanks a lot!
P

**Comment by zkamvar** [comment]
> As for setting the working directory for the activate.R script, I tried to use the open_deployer.rproj, but it didn't open the project and I ended up setting the working directory outside of a project. Normal?

No, this is not normal. When you say it didn't open the project, do you mean that RStudio opened, but not to the right project or RStudio did not open at all? If it's the second part, did you have RStudio installed?

> Beyond that, wonder if possible to add the **kableExtra** R package to the list of included packages? Or is there an alternative package recommended for creating tables in the deployer?

There's no reason why we can't include this.

> In regards to the activate.R script, do I download all the packages I'll need while in that working directory, or should they normally install from within any other script in other working directories after sourcing the activate.R?

Yes, you need to install all the packages you will need while in that working directory.

**Comment by pbkeating** [comment]
Hi Zhian,

Thanks for the rapid turnaround.


> > As for setting the working directory for the activate.R script, I tried to use the open_deployer.rproj, but it didn't open the project and I ended up setting the working directory outside of a project. Normal?
> 
> No, this is not normal. When you say it didn't open the project, do you mean that RStudio opened, but not to the right project or RStudio did not open at all? If it's the second part, did you have RStudio installed?

RStudio opens but not the R project, as in it says R project (none) in top right hand corner of R Studio.


> 
> > Beyond that, wonder if possible to add the **kableExtra** R package to the list of included packages? Or is there an alternative package recommended for creating tables in the deployer?
> 
> There's no reason why we can't include this.

Thanks!

> 
> > In regards to the activate.R script, do I download all the packages I'll need while in that working directory, or should they normally install from within any other script in other working directories after sourcing the activate.R?
> 
> Yes, you need to install all the packages you will need while in that working directory.

I think this would be good to clarify in the README.

**Comment by zkamvar** [comment]
> RStudio opens but not the R project, as in it says R project (none) in top right hand corner of R Studio.

Wait... you said that you clicked on a file called "open_deployer.rproj"? This file doesn't exist in this repo. There's `deployer.Rproj`, but not `open_deployer.Rproj`. Did you accidentally click on `update_deployer.R`?

**Comment by pbkeating** [comment]
Hmmm.. I got a copy of the deployer folder from a colleague here, perhaps could be the source of the confusion. See screenshot below
<img width="513" alt="Screen_shot_deployer_folder" src="https://user-images.githubusercontent.com/20238915/56136877-ed71cc00-5f93-11e9-930c-0fd4a8c48dd2.PNG">

**Comment by zkamvar** [comment]
:thinking: it was modified on the 7th, so perhaps your colleage changed things? 

In general, it's best to get a fresh copy of the deployer from the latest release (https://github.com/reconhub/deployer/releases/latest).

**Comment by pbkeating** [comment]
OK, thanks, Zhian!

Should I now close this? Who normally closes issues?

**Comment by zkamvar** [comment]
> OK, thanks, Zhian!
> 
> Should I now close this? Who normally closes issues?

Did not see this until today :scream_cat: 

The author of the issue should close it if they feel satisfied that the issue has been addressed, but in this case, I want to keep the issue open because I haven't yet made the changes (and I will also hijack it and add my own checklist of things that need clarification :grin:)

**Comment by zkamvar** [comment]
## Checklist of things that need to be clarified in the README of the deployer:

 - [x] Auto-update the README based on the current installed versions of the packages (it currently states R 3.5.1, which is innacurate
 - [ ] Revise intro paragraph to make clear that the user must still install the software to their computer.
 - [ ] State that installing R and RStudio will replace any existing versions
 - [ ] Make clear that opening the deployer project is only necessary when *installing* packages, but is not needed for loading/using packages. 
 - [ ] Create simple diagram demonstrating workflow to put at the top of the README
 - [ ] Document use of `update_package()` with caveats.
 - [ ] Clarify that the users should not modify the contents unless they are absolutely certain they know the consequences.
